### DISCUSSION TITLE
How to set Start_Time and Next_Fire_Time as the same in the qrtz_triggers tabl

### DISCUSSION THREAD

**Discussion by rineshsps** [quartznet/quartznet#1874] [Q&A]
How to set Start_Time and Next_Fire_Time as the same in the qrtz_triggers table? 

For the below code, Start_Time  638073099600000000, and Next_Fire_Time is coming at different time. 

```
var trigger = TriggerBuilder.Create()
                .WithIdentity("id123")
                .StartAt(startAt)
                .WithSchedule(CronScheduleBuilder
                    .DailyAtHourAndMinute(10, 10)) // hour, min
                .Build();
```

**Comment by lahma** [comment]
Why would you need that?

**Reply by rineshsps** [reply]
Hi @lahma

I'm using different timezone values, eg : utc +5.30, utc +4.30 etc, 

While saving time, Start_Time  and Next_Fire_Time values are different.  and the quartz is not executing as expected time.
So looking for a solution to resolve this. 


Internally it's updating those values, I want both values should be the same.

**Reply by lahma** [reply]
You shouldn't worry about internal storage details. Make sure that all `DateTime`s you provide are in UTC.

**Reply by rineshsps** [reply]
My users are in different time zones, so they want to choose the time accordingly to their local time on UI side. 
But while updating the values in the quartz side, start_time and next_fire_time values are saving different values. 

Here are the values saved in the table 

```
 "result": [
            {
                "schedName": "QuartzForTest",
                "triggerName": "24",
                "triggerGroup": "DEFAULT",
                "jobName": "job124",
                "jobGroup": "DEFAULT",
                "description": null,
                "nextFireTime": 638073315600000000,
                "prevFireTime": null,
                "priority": 5,
                "triggerState": "WAITING",
                "triggerType": "SIMPLE",
                "startTime": 638073099600000000,
                "endTime": 638075033990000000,
                "calendarName": null,
                "misfireInstr": 0,
                "jobData": null,
                "qrtzJobDetails": null,
                "qrtzBlobTriggers": null,
                "qrtzCronTriggers": null,
                "qrtzSimpleTriggers": null,
                "qrtzSimpropTriggers": null
            }
        ]
```

**Reply by lahma** [reply]
Well startAt tells where from to start counting from, say 3PM , if cron is to be run at 4PM every day then start and next fire time will be different.

**Reply by rineshsps** [reply]
Oke,

For eg : consider the current time is 2 PM and I want to run the cron every 1 hour at 2.30PM from today onwards.
I'll set the start time as 2.30 PM.

Then the occurrence could be 2.30, 3.30, 4.30 etc right? In this case start and next fire will be the same 
ie 2.30 PM or will it differ start time to 2 PM and the next fire is 2.30 PM ?

**Reply by lahma** [reply]
If you set the start time as the same time as first calculated fire they should be quite the same, I think. Unless it's calculated as next as it's exactly the same. Please experiment!
